src/hooks/useExecuteGetFeature.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  useCallback
+} from 'react';
+
+import {
+  FeatureCollection
+} from 'geojson';
+
+import {
+  equalTo
+} from 'ol/format/filter';
+
+import OlFormatWFS from 'ol/format/WFS';
+import OlSourceImageWMS from 'ol/source/ImageWMS';
+import OlSourceTileWMS from 'ol/source/TileWMS';
+
+import useMap from '@terrestris/react-geo/dist/Hook/useMap';
+import {
+  WmsLayer
+} from '@terrestris/react-geo/dist/Util/typeUtils';
+
+import {
+  getBearerTokenHeader
+} from '@terrestris/shogun-util/dist/security/getBearerTokenHeader';
+
+import useExecuteWfsDescribeFeatureType from './useExecuteWfsDescribeFeatureType';
+import useSHOGunAPIClient from './useSHOGunAPIClient';
+
+export type GetFeatureOpts = {
+  layer: WmsLayer;
+  // TODO Pass filter instead of featureId
+  featureId: string;
+};
+
+export const useExecuteGetFeature = () => {
+  const client = useSHOGunAPIClient();
+  const map = useMap();
+  const executeWfsDescribeFeatureType = useExecuteWfsDescribeFeatureType();
+
+  const executeGetFeature = useCallback(async (opts: GetFeatureOpts) => {
+    if (!map) {
+      return;
+    }
+
+    if (!opts.featureId) {
+      return;
+    }
+
+    const describeFeatureType = await executeWfsDescribeFeatureType(opts.layer);
+
+    if (!describeFeatureType) {
+      return;
+    }
+
+    let url;
+
+    const source = opts.layer.getSource();
+    if (source instanceof OlSourceImageWMS) {
+      url = (source as OlSourceImageWMS).getUrl();
+    }
+    if (source instanceof OlSourceTileWMS) {
+      const urls = (source as OlSourceTileWMS).getUrls();
+      url = urls ? urls[0] : undefined;
+    }
+
+    if (!url) {
+      return;
+    }
+
+    if (url.endsWith('?')) {
+      url = url.slice(0, -1);
+    }
+
+    const featureRequest = new OlFormatWFS().writeGetFeature({
+      srsName: map.getView().getProjection().getCode(),
+      featureNS: describeFeatureType.targetNamespace,
+      featurePrefix: describeFeatureType.targetPrefix,
+      featureTypes: [source?.getParams().LAYERS],
+      outputFormat: 'application/json',
```

Comment:
Will always GeoServer be used? Not all WFS servers support `application/json` as output…

Reply:
Internally GeoServer will always be used, but actually it's possible to include layers from external servers as well which might be used by the given method and might fail. Good point.

@LukasLohoff What do you think, should we switch to GML here instead?

Reply:
good point @ahennr 
we'll test if we can switch to GML here

Reply:
we had some issues parsing the GML with openlayers. For now, we'll use JSON and may implement this in the future